src/NServiceBus.AcceptanceTests/Routing/When_using_instance_ids.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace NServiceBus.AcceptanceTests.ScaleOut
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+
+    public class When_using_instance_ids : NServiceBusAcceptanceTest
+    {
+        public static string GetReceiverEndpoint()
+        {
+            return AcceptanceTesting.Customization.Conventions.EndpointNamingConvention(typeof(Receiver));
+        }
+
+
+        [Test]
+        public async Task Should_be_addressable_both_by_shared_queue_and_specific_queue()
+        {
+            var context = await Scenario.Define<Context>()
+                    .WithEndpoint<Receiver>()
+                    .WithEndpoint<UnawareSender>(b => b.When(s => s.Send(new MyMessage())))
+                    .WithEndpoint<AwareSender>(b => b.When(s => s.Send(new MyMessage())))
+                    .Done(c => c.MessagesReceived > 1)
+                    .Run();
+
+            Assert.AreEqual(2, context.MessagesReceived);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public int MessagesReceived;
+        }
+
+        public class UnawareSender : EndpointConfigurationBuilder
+        {
+            public UnawareSender()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.Routing().UnicastRoutingTable.RouteToEndpoint(typeof(MyMessage), GetReceiverEndpoint());
+                });
+            }
+        }
+
+        public class AwareSender : EndpointConfigurationBuilder
+        {
+
+            public AwareSender()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.Routing().UnicastRoutingTable.RouteToEndpoint(typeof(MyMessage), GetReceiverEndpoint());
```

Comment:
Yes. L54 is the logical routing (type -> endpoint). L55 is physical (endpoint->list of instances)